File: UIMA.Ext.Corpus.Statistics/src/main/java/ru/kfu/itis/issst/corpus/statistics/dao/CorpusDAO.java
package ru.kfu.itis.issst.corpus.statistics.dao;

import java.io.IOException;
import java.net.URI;
import java.net.URISyntaxException;
import java.util.Set;

import org.apache.uima.cas.CAS;
import org.xml.sax.SAXException;

public interface CorpusDAO {

	Set<URI> getDocuments() throws URISyntaxException;

	Set<String> getAnnotatorIds(URI docURI) throws IOException;

	void getDocumentCas(URI docURI, String annotatorId, CAS aCAS)
			throws IOException, SAXException;
}

File: UIMA.Ext.Corpus.Util/src/main/scala/ru/kfu/itis/issst/corpus/dao/CorpusDAO.scala
/**
 *
 */
package ru.kfu.itis.issst.corpus.dao

import java.net.URI
import org.apache.uima.cas.CAS

/**
 * @author Rinat Gareev (Kazan Federal University)
 *
 */
trait CorpusDAO extends ru.kfu.itis.issst.corpus.statistics.dao.corpus.CorpusDAO {

  def persist(docUri: URI, annotatorId: String, cas: CAS)

}